flink-tests/src/test/java/org/apache/flink/test/misc/CustomSerializationITCase.java

```diff
@@ -62,127 +68,103 @@ public static Configuration getConfiguration() {
-
-			env.execute();
-		}
-		catch (JobExecutionException e) {
-			Throwable rootCause = e.getCause();
-			assertTrue(rootCause instanceof IOException);
-			assertTrue(rootCause.getMessage().contains("broken serialization"));
-		}
-		catch (Exception e) {
-			e.printStackTrace();
-			fail(e.getMessage());
-		}
-	}
+	public void testIncorrectSerializer3() throws Exception {
+		expectedException.expect(JobExecutionException.class);
+		expectedException.expectCause(
+			Matchers.both(isA(IOException.class))
+				.and(hasProperty("message", containsString("broken serialization"))));
 
-	@Test
-	public void testIncorrectSerializer3() {
-		try {
-			ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
-			env.setParallelism(PARLLELISM);
-			env.getConfig().disableSysoutLogging();
-
-			env
-					.generateSequence(1, 10 * PARLLELISM)
-					.map(new MapFunction<Long, ConsumesTooLittle>() {
-						@Override
-						public ConsumesTooLittle map(Long value) throws Exception {
-							return new ConsumesTooLittle();
-						}
-					})
-					.rebalance()
-					.output(new DiscardingOutputFormat<ConsumesTooLittle>());
-
-			env.execute();
-		}
-		catch (JobExecutionException e) {
-			Throwable rootCause = e.getCause();
-			assertTrue(rootCause instanceof IOException);
-			assertTrue(rootCause.getMessage().contains("broken serialization"));
-		}
-		catch (Exception e) {
-			e.printStackTrace();
-			fail(e.getMessage());
-		}
+		ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+		env.setParallelism(PARLLELISM);
+		env.getConfig().disableSysoutLogging();
+
+		env
+				.generateSequence(1, 10 * PARLLELISM)
+				.map(new MapFunction<Long, ConsumesTooLittle>() {
+					@Override
+					public ConsumesTooLittle map(Long value) throws Exception {
+						return new ConsumesTooLittle();
+					}
+				})
+				.rebalance()
+				.output(new DiscardingOutputFormat<ConsumesTooLittle>());
+
+		env.execute();
 	}
 
 	@Test
-	public void testIncorrectSerializer4() {
-		try {
-			ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
-			env.setParallelism(PARLLELISM);
-			env.getConfig().disableSysoutLogging();
-
-			env
-					.generateSequence(1, 10 * PARLLELISM)
-					.map(new MapFunction<Long, ConsumesTooLittleSpanning>() {
-						@Override
-						public ConsumesTooLittleSpanning map(Long value) throws Exception {
-							return new ConsumesTooLittleSpanning();
-						}
-					})
-					.rebalance()
-					.output(new DiscardingOutputFormat<ConsumesTooLittleSpanning>());
-
-			env.execute();
-		}
-		catch (ProgramInvocationException e) {
-			Throwable rootCause = e.getCause().getCause();
-			assertTrue(rootCause instanceof IOException);
-			assertTrue(rootCause.getMessage().contains("broken serialization"));
-		}
-		catch (Exception e) {
-			e.printStackTrace();
-			fail(e.getMessage());
-		}
+	public void testIncorrectSerializer4() throws Exception {
```

Comment:
There are two many duplicate codes in above four tests, and only the map function is different.
Is it necessary or feasible to further deduplicate these common codes?

Reply:
I actually missed this one, sorry
-> it was actually easy to fix